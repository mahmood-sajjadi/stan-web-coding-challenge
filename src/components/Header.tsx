import { useMemo } from "react";
import { Button } from "./Button";

type Props = {
    children?: React.ReactNode;
    headerStyle?: React.CSSProperties;
    headerMainStyle?: React.CSSProperties;
    headerPageStyle?: React.CSSProperties;
}

const headerStyleBase = {
    display: 'box',
    width: '100%',
}

const headerMainStyleBase = {
    display: 'box',
    width: '100%',
    height: '30px',
    paddingTop: '10px',
    paddingBottom: '10px',
    background: 'linear-gradient(0deg,rgb(46, 149, 250) 0%, rgb(18, 100, 246) 100%)',
    boxShadow: '0px 3px 3px rgb(45, 45, 45)',
    position: 'relative',
    zIndex: '10',
}

const headerCTAHolderStyle = {
    paddingTop: '10px',
    paddingBottom: '10px',
    fontSize: '10px',
    float: 'right'
}

const loginStyle = {
    marginRight: '10px',
}

const headerPageStyleBase = {
    paddingTop: '10px',
    paddingBottom: '10px',
    display: 'box',
    width: '100%',
    height: '30px',
    background: 'rgb(65, 65, 65)'
}

export function Header(props: Props) {
    const { headerStyle, headerMainStyle, headerPageStyle, children } = props;

    const headerStyleCalculated = useMemo(() => ({
        ...headerStyleBase,
        ...headerStyle,
    }), [headerStyle]);

    const headerMainStyleCalculated = useMemo(() => ({
        ...headerMainStyleBase,
        ...headerMainStyle,
    }), [headerMainStyle]);

    const headerPageStyleCalculated = useMemo(() => ({
        ...headerPageStyleBase,
        ...headerPageStyle,
    }), [headerPageStyle]);

    return (
        <header style={headerStyleCalculated}>
            <div style={headerMainStyleCalculated}>
                DEMO Streaming
                <div style={headerCTAHolderStyle}>
                    <span style={loginStyle}>Log in</span>
                    <Button>Start your free trial</Button>
                </div>
            </div>
            <div style={headerPageStyleCalculated}>
                {children}
            </div>
        </header>
    );
}