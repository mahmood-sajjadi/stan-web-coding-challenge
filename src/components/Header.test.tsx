import { render, screen } from '@testing-library/react'
import {Header} from './Header'

describe('Header', () => {
  it('renders the Header component', async () => {
    const content = 'test header text';
    const header = render(
        <Header>
            {content}
        </Header>
    );
    const contentResult = await screen.findByText(content);
    expect(header).toMatchSnapshot();
    expect(contentResult).toBeTruthy()
  })
})