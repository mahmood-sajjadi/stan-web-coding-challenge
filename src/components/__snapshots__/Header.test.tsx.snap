// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > renders the Header component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        style="display: box; width: 100%;"
      >
        <div
          style="display: box; width: 100%; height: 30px; padding-top: 10px; padding-bottom: 10px; background: linear-gradient(0deg,rgb(46, 149, 250) 0%, rgb(18, 100, 246) 100%); box-shadow: 0px 3px 3px rgb(45, 45, 45); position: relative; z-index: 10;"
        >
          DEMO Streaming
          <div
            style="padding-top: 10px; padding-bottom: 10px; font-size: 10px; float: right;"
          >
            <span
              style="margin-right: 10px;"
            >
              Log in
            </span>
            <div
              style="color: white; background: linear-gradient(0deg,rgb(68, 68, 68) 0%, rgb(51, 51, 51) 100%); display: inline; padding: 10px;"
            >
              Start your free trial
            </div>
          </div>
        </div>
        <div
          style="padding-top: 10px; padding-bottom: 10px; display: box; width: 100%; height: 30px; background: rgb(65, 65, 65);"
        >
          test header text
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      style="display: box; width: 100%;"
    >
      <div
        style="display: box; width: 100%; height: 30px; padding-top: 10px; padding-bottom: 10px; background: linear-gradient(0deg,rgb(46, 149, 250) 0%, rgb(18, 100, 246) 100%); box-shadow: 0px 3px 3px rgb(45, 45, 45); position: relative; z-index: 10;"
      >
        DEMO Streaming
        <div
          style="padding-top: 10px; padding-bottom: 10px; font-size: 10px; float: right;"
        >
          <span
            style="margin-right: 10px;"
          >
            Log in
          </span>
          <div
            style="color: white; background: linear-gradient(0deg,rgb(68, 68, 68) 0%, rgb(51, 51, 51) 100%); display: inline; padding: 10px;"
          >
            Start your free trial
          </div>
        </div>
      </div>
      <div
        style="padding-top: 10px; padding-bottom: 10px; display: box; width: 100%; height: 30px; background: rgb(65, 65, 65);"
      >
        test header text
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
